packages/image_picker/image_picker/example/ios/RunnerTests/PickerSaveImageToPathOperationTests.m

```diff
@@ -0,0 +1,85 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <OCMock/OCMock.h>
+#import <UniformTypeIdentifiers/UniformTypeIdentifiers.h>
+#import <XCTest/XCTest.h>
+
+@import image_picker;
+
+@interface PickerSaveImageToPathOperationTests : XCTestCase
+
+@end
+
+@implementation PickerSaveImageToPathOperationTests
+
+- (void)testSaveWebPImage API_AVAILABLE(ios(14)) {
+  NSURL *imageURL = [[NSBundle bundleForClass:[self class]] URLForResource:@"webpImage"
+                                                             withExtension:@"webp"];
+  NSItemProvider *itemProvider = [[NSItemProvider alloc] initWithContentsOfURL:imageURL];
+  PHPickerResult *result = [self createPickerResult:itemProvider
+                                     withIdentifier:UTTypeWebP.identifier];
+
+  [self saveImageToPath:result];
+}
+
+- (void)testSavePNGImage API_AVAILABLE(ios(14)) {
+  NSURL *imageURL = [[NSBundle bundleForClass:[self class]] URLForResource:@"pngImage"
+                                                             withExtension:@"png"];
+  NSItemProvider *itemProvider = [[NSItemProvider alloc] initWithContentsOfURL:imageURL];
+  PHPickerResult *result = [self createPickerResult:itemProvider
+                                     withIdentifier:UTTypeWebP.identifier];
+
+  [self saveImageToPath:result];
+}
+
+- (void)testSaveJPGImage API_AVAILABLE(ios(14)) {
+  NSURL *imageURL = [[NSBundle bundleForClass:[self class]] URLForResource:@"jpgImage"
+                                                             withExtension:@"jpg"];
+  NSItemProvider *itemProvider = [[NSItemProvider alloc] initWithContentsOfURL:imageURL];
+  PHPickerResult *result = [self createPickerResult:itemProvider
+                                     withIdentifier:UTTypeWebP.identifier];
+
+  [self saveImageToPath:result];
+}
+
+- (void)testSaveGIFImage API_AVAILABLE(ios(14)) {
+  NSURL *imageURL = [[NSBundle bundleForClass:[self class]] URLForResource:@"gifImage"
+                                                             withExtension:@"gif"];
+  NSItemProvider *itemProvider = [[NSItemProvider alloc] initWithContentsOfURL:imageURL];
+  PHPickerResult *result = [self createPickerResult:itemProvider
+                                     withIdentifier:UTTypeWebP.identifier];
+
+  [self saveImageToPath:result];
+}
+
+- (PHPickerResult *)createPickerResult:(NSItemProvider *)itemProvider
+                        withIdentifier:(NSString *)identifier API_AVAILABLE(ios(14)) {
+  PHPickerResult *result = OCMClassMock([PHPickerResult class]);
+
+  OCMStub([result itemProvider]).andReturn(itemProvider);
+  OCMStub([result assetIdentifier]).andReturn(identifier);
+
+  return result;
+}
+
+- (void)saveImageToPath:(PHPickerResult *)result API_AVAILABLE(ios(14)) {
```

Comment:
nits:
This method does more than just "saveImageToPath", this is actually testing the save image operation.
How about rename it to something like: `testSaveImageOperationWithResult:`.

Reply:
It can't start with "test", but something similar with "verify" or "validate" would be good.